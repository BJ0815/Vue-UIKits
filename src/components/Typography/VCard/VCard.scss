@import './variables.scss';

%ellipsis {
  overflow: hidden;
  text-overflow: ellipsis;

  /* Required for text-overflow to do anything */
  white-space: nowrap;
}

%clamp {
  display: -webkit-box;

  /* Required for text-overflow to do anything */
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: normal;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

// Block
.v-card {
  position: relative;
  display: block;
  min-width: 320px;
  max-width: 100%;
  text-decoration: none;
  white-space: $card-white-space;
  color: rgba(0, 0, 0, 0.87);
  background-color: #fff;
  border-width: $card-outlined-border-width;
  border-radius: 4px;
  outline: none;
  box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.2), 0 4px 5px 0 rgba(0, 0, 0, 0.14),
    0 1px 10px 0 rgba(0, 0, 0, 0.12);
  transition-property: box-shadow, opacity;
  overflow-wrap: $card-overflow-wrap;
}

.v-card__title {
  display: flex;
  align-items: center;
  padding: $card-title-padding;
  font-size: $card-title-font-size;
  font-weight: $card-title-font-weight;
  flex-wrap: wrap;
  letter-spacing: $card-title-letter-spacing;
  line-height: $card-title-line-height;
  word-break: $card-title-word-break;
}

.v-card__footer {
  display: flex;
  align-items: center;
  padding: $card-actions-padding;
}

.v-card__text {
  padding: $card-actions-padding;
  font-size: $card-text-font-size;
  font-weight: $card-text-font-weight;
  text-align: left;
  line-height: $card-text-line-height;
  letter-spacing: $card-text-letter-spacing;
  * {
    @extend %clamp;
  }
}
