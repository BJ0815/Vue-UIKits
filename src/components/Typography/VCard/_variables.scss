$card-outlined-border-width: thin !default;
$card-overflow-wrap: break-word !default;
$card-white-space: normal !default;

$card-title-font-size: 1.25rem !default;
$card-title-font-weight: 500 !default;
$card-title-letter-spacing: .0125em !default;
$card-title-line-height: 2rem !default;
$card-title-word-break: break-all !default;
$card-title-padding: 10px !default;

$card-actions-padding: 10px !default;

$card-text-font-size: 1rem !default;
$card-text-font-weight: 400 !default;
$card-text-letter-spacing: .0125em !default;
$card-text-line-height: 1.2rem !default;