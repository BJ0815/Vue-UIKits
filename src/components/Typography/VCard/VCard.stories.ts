import { storiesOf } from '@storybook/vue';
import { action } from '@storybook/addon-actions';

import VCard, { VCardFooter, VCardTitle, VCardText } from './index';

storiesOf('VCard', module).add('default', () => ({
  components: {
    VCard,
    VCardFooter,
    VCardTitle,
    VCardText,
  },
  props: {
    tag: {
      type: String,
      default: () => 'div',
    },
  },
  template: `
  <VCard :tag="tag" class="class1" style="color: black;" @click="onClick">
    <VCardTitle>
      <h1>Hello~</h1>
    </VCardTitle>
    <VCardText>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Eveniet neque enim quibusdam
        illum delectus quas, deserunt voluptatem libero vero officia molestiae, ipsa aut
        aperiam, at doloremque alias rem quasi dolores!
      </p>
    </VCardText>
    <VCardFooter>
      <p>Footer</p>
    </VCardFooter>
  </VCard>`,
  methods: {
    onClick: action('click'),
  },
}));
