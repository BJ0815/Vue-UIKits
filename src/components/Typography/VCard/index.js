// Styles
import './VCard.scss';
// Types
import Vue from 'vue';

export function createSimpleFunctional(c, el = 'div', name) {
  return Vue.extend({
    name: name || c.replace(/__/g, '-'),
    functional: true,
    render(h, { data, children }) {
      const myData = data;
      myData.staticClass = `${c} ${data.staticClass || ''}`.trim();
      return h(el, myData, children);
    },
  });
}
/**
 * Card Utils Components
 */
export const VCardTitle = createSimpleFunctional('v-card__title');
export const VCardFooter = createSimpleFunctional('v-card__footer');
export const VCardText = createSimpleFunctional('v-card__text');
/**
 * Card UI Component
 */
export default Vue.extend({
  name: 'v-card',
  props: {
    tag: {
      type: String,
      default: 'div',
    },
  },
  computed: {
    classes() {
      return ['v-card'];
    },
    styles() {
      return { color: 'red' };
    },
  },
  render(h) {
    const data = {
      attrs: {
        tabindex: 'tabindex' in this.$attrs ? this.$attrs.tabindex : undefined,
      },
      class: this.classes,
      style: this.styles,
      on: this.$listeners,
      ref: 'card',
    };
    // <div><slot /></div>
    return h(this.tag, data, [this.$slots.default]);
  },
});
