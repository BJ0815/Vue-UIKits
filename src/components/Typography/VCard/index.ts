// Styles
import './VCard.scss';

// Types
import Vue, { VNode, VNodeData } from 'vue';

export function createSimpleFunctional(c: string, el = 'div', name?: string) {
  return Vue.extend({
    name: name || c.replace(/__/g, '-'),

    functional: true,

    render(h, { data, children }): VNode {
      const myData = data;
      myData.staticClass = `${c} ${data.staticClass || ''}`.trim();

      return h(el, myData, children);
    },
  });
}

/**
 * Card Utils Components
 */
export const VCardTitle = createSimpleFunctional('v-card__title');
export const VCardFooter = createSimpleFunctional('v-card__footer');
export const VCardText = createSimpleFunctional('v-card__text');

/**
 * Card UI Component
 */
export default Vue.extend({
  name: 'v-card',
  props: {
    tag: {
      type: String,
      default: () => 'div',
    },
  },
  computed: {
    classes() {
      return ['v-card'];
    },
    styles() {
      return { color: 'red' };
    },
  },
  render(h): VNode {
    const data: VNodeData = {
      attrs: {
        tabindex: 'tabindex' in this.$attrs ? this.$attrs.tabindex : undefined,
      },
      class: this.classes,
      style: this.styles,
      on: this.$listeners,
      ref: 'card',
    };
    // <div><slot /></div>
    return h(this.tag, data, [this.$slots.default]);
  },
});
