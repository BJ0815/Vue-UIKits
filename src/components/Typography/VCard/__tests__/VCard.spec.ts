import {
  mount, Wrapper, MountOptions,
} from '@vue/test-utils';
import Vue from 'vue';
import VCard from '../index';

describe('HelloWorld.vue', () => {
  let mountFunction: (options?: MountOptions<Vue>) => Wrapper<Vue>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Vue>) => mount(VCard, {
      ...options,
    });
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
