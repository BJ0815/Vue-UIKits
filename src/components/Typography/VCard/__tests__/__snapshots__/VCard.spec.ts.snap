// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue should render component and match snapshot 1`] = `<div class="v-card" style="color: red;"></div>`;
